Remove GNU_SOURCE definition rom the only file that has it.

--- ./common/smtppass.c	2014-07-06 12:26:26.000000000 +0200
+++ ./common/smtppass.c	2014-07-06 12:26:52.367104112 +0200
@@ -38,8 +38,6 @@
  *  Olivier Beyssac <ob@r14.freenix.org>
  */ 
 
-#define _GNU_SOURCE
-
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/socket.h>
